pkg/mtbroker/filter/filter_handler.go

```diff
@@ -278,20 +270,30 @@ func (h *Handler) writeResponse(ctx context.Context, writer http.ResponseWriter,
 		n, _ := response.BodyReader.Read(body)
 		response.BodyReader.Close()
 		if n != 0 {
+			// Note that we could just use StatusInternalServerError, but to distinguish
+			// between the two failure cases, we use a different code here.
+			writer.WriteHeader(http.StatusBadGateway)
 			return resp.StatusCode, errors.New("received a non-empty response not recognized as CloudEvent. The response MUST be or empty or a valid CloudEvent")
 		}
 		h.logger.Debug("Response doesn't contain a CloudEvent, replying with an empty response", zap.Any("target", target))
+		writer.WriteHeader(resp.StatusCode)
 		return resp.StatusCode, nil
 	}
 
 	event, err := binding.ToEvent(ctx, response)
 	if err != nil {
+		// Note that we could just use StatusInternalServerError, but to distinguish
+		// between the two failure cases, we use a different code here.
+		writer.WriteHeader(http.StatusBadGateway)
 		// Malformed event, reply with err
 		return resp.StatusCode, err
 	}
 
 	// Reattach the TTL (with the same value) to the response event before sending it to the Broker.
 	if err := broker.SetTTL(event.Context, ttl); err != nil {
+		// Note that we could just use StatusInternalServerError, but to distinguish
```

Comment:
This comment should be adjusted since the code below is StatusInternalServerError